integrations/philip-click-up/click_up/client.py

```diff
@@ -0,0 +1,116 @@
+from port_ocean.utils import http_async_client
+from httpx import Timeout
+from typing import AsyncGenerator, Any
+from datetime import datetime
+
+
+class ClickUpClient:
+
+    def __init__(self, click_up_token: str) -> None:
+
+        self.click_up_api_url = "https://api.clickup.com/api/v2"
+        self.click_up_token = click_up_token
+        self.client = http_async_client
+        self.client.headers = {
+            "Authorization": self.click_up_token,
+            **self.client.headers,
+        }
+        self.client.timeout = Timeout(30)
+
+    def milesecond_time_to_date_str(self, data: dict):
+
+        if data.get("date_created"):
+            date_created_ms = int(data["date_created"]) / 1000
+
+            data["date_created"] = datetime.fromtimestamp(timestamp=date_created_ms)
+
+        if data.get("date_updated"):
+            date_updated_ms = int(data["date_updated"]) / 1000
+
+            data["date_updated"] = datetime.fromtimestamp(timestamp=date_updated_ms)
+
+        return data
+
+    async def __generate_teams(self):
+
+        response = await self.client.get(url=f"{self.click_up_api_url}/team")
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def __get_spaces(self, team_id: int):
+
+        response = await self.client.get(url=f"{self.click_up_api_url}/{team_id}/space")
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def __get_folder(self, space_id: str):
+
+        response = await self.client.get(
+            url=f"{self.click_up_api_url}/space/{space_id}/folder"
+        )
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def __get_projects(self, folder_id: int):
+        response = await self.client.get(
+            url=f"{self.click_up_api_url}/folder/{folder_id}/list"
+        )
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def __get_issues(self, list_id: int):
+
+        response = await self.client.get(url=f"{self.click_up_api_url}/{list_id}/space")
```

Comment:
Are you sure this endpoint will return issues? This will return 404 since it doesn't exist

Reply:
This was an oversight. I have fixed this.